src/Files.App/ViewModels/UserControls/StatusCenterViewModel.cs

```diff
@@ -0,0 +1,140 @@
+		public int InfoBadgeState
+		{
+			get
+			{
+				var anyFailure = StatusBannersSource.Any(i => i.Status != ReturnResult.InProgress && i.Status != ReturnResult.Success);
+
+				return (anyFailure, AnyOperationsOngoing) switch
+				{
+					(false, false) => 0, // All success
+					(false, true) => 1,  // Ongoing
+					(true, true) => 2,   // Ongoing with failure
+					(true, false) => 3   // Completed with failure
+				};
+			}
+		}
+
+		public int InfoBadgeValue
+		{
+			get => OngoingOperationsCount > 0 ? OngoingOperationsCount : -1;
+		}
+
+		public event EventHandler<PostedStatusBanner> ProgressBannerPosted;
+
+		public OngoingTasksViewModel()
+		{
+			StatusBannersSource.CollectionChanged += (s, e) => OnPropertyChanged(nameof(AnyBannersPresent));
+		}
+
+		public PostedStatusBanner PostBanner(string title, string message, int initialProgress, ReturnResult status, FileOperationType operation)
+		{
+			StatusBanner banner = new StatusBanner(message, title, initialProgress, status, operation);
+			PostedStatusBanner postedBanner = new PostedStatusBanner(banner, this);
+
+			StatusBannersSource.Insert(0, banner);
+			ProgressBannerPosted?.Invoke(this, postedBanner);
+
+			UpdateBanner(banner);
+
+			return postedBanner;
+		}
+
+		public PostedStatusBanner PostOperationBanner(string title, string message, int initialProgress, ReturnResult status, FileOperationType operation, CancellationTokenSource cancellationTokenSource)
+		{
+			StatusBanner banner = new(message, title, initialProgress, status, operation)
+			{
+				CancellationTokenSource = cancellationTokenSource,
+			};
+
+			PostedStatusBanner postedBanner = new(banner, this, cancellationTokenSource);
+
+			StatusBannersSource.Insert(0, banner);
+			ProgressBannerPosted?.Invoke(this, postedBanner);
+
+			UpdateBanner(banner);
+
+			return postedBanner;
+		}
+
+		public PostedStatusBanner PostActionBanner(string title, string message, string primaryButtonText, string cancelButtonText, Action primaryAction)
+		{
+			StatusBanner banner = new(message, title, primaryButtonText, cancelButtonText, primaryAction);
+			PostedStatusBanner postedBanner = new(banner, this);
+
+			StatusBannersSource.Insert(0, banner);
+			ProgressBannerPosted?.Invoke(this, postedBanner);
+
+			UpdateBanner(banner);
+
+			return postedBanner;
+		}
+
+		public bool CloseBanner(StatusBanner banner)
+		{
+			if (!StatusBannersSource.Contains(banner))
+				return false;
+
+			StatusBannersSource.Remove(banner);
+
+			UpdateBanner(banner);
+
+			return true;
+		}
+
+		public void UpdateBanner(StatusBanner banner)
+		{
+			OnPropertyChanged(nameof(OngoingOperationsCount));
+			OnPropertyChanged(nameof(AnyOperationsOngoing));
+			OnPropertyChanged(nameof(InfoBadgeState));
+			OnPropertyChanged(nameof(InfoBadgeValue));
+		}
+
+		public void UpdateMedianProgress()
+		{
+			if (AnyOperationsOngoing)
+				MedianOperationProgressValue = (int)StatusBannersSource.Where((item) => item.IsProgressing).Average(x => x.Progress);
+		}
```

Comment:
I know this might be out of scope, but I think we should rename `MedianOperationProgressValue` to `AverageOperationProgressValue`. We are calculating the average value not the median one